tests/test_env.py

```diff
@@ -1,20 +1,86 @@
+        "[pytest]\nenv = MAGIC=alpha",
+        {"MAGIC": "alpha"},
+        id="When ini has a new key, Then it is added to env"
+    ),
+    pytest.param(
+        {"MAGIC": "alpha"},
+        "[pytest]\nenv = MAGIC=beta",
+        {"MAGIC": "beta"},
+        id="When ini has a key which already exists in env, Then the env value is overwritten"
+    ),
+    pytest.param(
+        {"MAGIC": "alpha"},
+        "[pytest]\nenv = D:MAGIC=beta",
+        {"MAGIC": "alpha"},
+        id="Given D flag, When ini has a key which already exists in env, Then the original env value is kept"
+    ),
+    pytest.param(
+        {"PLANET": "world"},
+        "[pytest]\nenv = MAGIC=hello_{PLANET}",
+        {"MAGIC": "hello_world"},
+        id="When ini value has an env key between curly braces, Then the key's value is interpolated"
+    ),
+    pytest.param(
+        {"PLANET": "world"},
+        "[pytest]\nenv = R:MAGIC=hello_{PLANET}",
+        {"MAGIC": "hello_{PLANET}"},
+        id="When ini key has R flag, Then ini value is not interpolated"
+    ),
+    pytest.param(
+        {"MAGIC": "a"},
+        "[pytest]\nenv = R:MAGIC={MAGIC}b\n D:MAGIC={MAGIC}c\n MAGIC={MAGIC}d",
+        {"MAGIC": "{MAGIC}bd"},
+        id="When ini has repeating keys, Then values and flags are evaluated separately and incrementally"
+    ),
+    pytest.param(
+        {"MAGIC": "alpha"},
+        "[pytest]\nenv = D:R:MAGIC=beta",
+        {"MAGIC": "alpha"},
+        id="When ini key has two flags, Then both are applied"
+    ),
+    pytest.param(
+        {"MAGIC": "alpha"},
+        "[pytest]\nenv = R:D:MAGIC=beta",
+        {"MAGIC": "alpha"},
+        id="When ini key has two flags (in reverse order), Then both are applied"
+    ),
+])
+def test_cases(testdir: pytest.Testdir, existing_env_vars: Dict[str, str], ini_contents: str,
+               expected_env_vars: Dict[str, str]) -> None:
+    set_env_variables(testdir, existing_env_vars)
+    write_to_files(testdir, ini_contents, expected_env_vars)
+    result = testdir.runpytest()
+    result.assert_outcomes(passed=1)
 
 
-def test_simple(example: pytest.Testdir) -> None:
-    (example.tmpdir / "pytest.ini").write_text("[pytest]\nenv = MAGIC=alpha", encoding="utf-8")
-    example.monkeypatch.setenv("_PATCH", "alpha")
-    result = example.runpytest()
-    result.assert_outcomes(passed=1)
+def set_env_variables(testdir: pytest.Testdir, env_vars: Dict[str, str]):
+    for env_var, val in env_vars.items():
+        testdir.monkeypatch.setenv(env_var, val)
+
+
+def write_to_files(testdir: pytest.Testdir, ini_contents: str,
+                   expected_env_vars: Dict[str, str]) -> None:
+    contents = """
+from __future__ import annotations
+import ast
+import os
+
+
+def test_() -> None:
+    expected_env_vars = ast.literal_eval(\"\"\"%s\"\"\")
+    for key, expected_val in expected_env_vars.items():
+        assert os.environ[key] == expected_val, f'Assertion failed for key {key}'
+    """ % expected_env_vars
+
```

Comment:
Please never use % for string formatting, that's a Python 2.6 thing that should die out. Always use f-strings in Python 3. Also I don't like such strings in tests because they are not debugable and coverage-able, this is why I had the example.py in the first place.

Reply:
> Can you please allow change from maintainers? I can't push to this PR [gaborbernat@7c193da](https://github.com/gaborbernat/pytest-env/commit/7c193da4420ceab55b12502821082e5ecefa4a1c)

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

Reply:
> Also I don't like such strings in tests because they are not debugable and coverage-able

How could we solve this? I now put the tests into files, just as in the original example.py

Reply:
> > Can you please allow change from maintainers? I can't push to this PR [gaborbernat@7c193da](https://github.com/gaborbernat/pytest-env/commit/7c193da4420ceab55b12502821082e5ecefa4a1c)
> 
> https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

It is already enabled for this PR (#9 ), let me know if I can do anything else
<img width="1000" alt="image" src="https://user-images.githubusercontent.com/4084885/197397215-1115730b-f37e-44cb-8d9d-c1dc44b51f82.png">